onedal/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,99 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+
+from daal4py.sklearn._utils import get_dtype
+
+from ..datatypes import _convert_to_supported, from_table, to_table
+from ..utils import _check_array
+from .pca import BasePCA
+
+
+class IncrementalPCA(BasePCA):
+
+    def __init__(
+        self,
+        n_components=None,
+        is_deterministic=True,
+        method="cov",
```

Comment:
seems these cannot be set from sklearnex API - is it meant to always be the same/default?

Reply:
I think currently it is, until ability to set the method will be added to sklearn IncrementalPCA. Moreover, for GPU backend it is the only available method now to compute incremental PCA on C++ side

Reply:
@md-shafiul-alam any thoughts? sounds good to me

Reply:
scikit-learn uses "svd". In our vanilla PCA, the method has been recently updated to fall back to sklearn-svd starting from sklearn 1.5 until oneDAL "svd" is optimized. Separate investigation will be done soon. For the purpose of this PR, I am fine with the "cov", also because the GPU support is limited to "cov" in oneDAL incremental PCA.